deltacat/tests/compactor/test_compaction_session.py

```diff
@@ -0,0 +1,406 @@
+    ],
+    [
+        (
+            test_name,
+            primary_keys,
+            in_sort_keys,
+            in_partition_keys,
+            column_names,
+            arrow_arrays,
+            rebase_source_partition_locator,
+            partition_values,
+            expected_result,
+            validation_callback_func,
+        )
+        for test_name, (
+            primary_keys,
+            in_sort_keys,
+            in_partition_keys,
+            column_names,
+            arrow_arrays,
+            rebase_source_partition_locator,
+            partition_values,
+            expected_result,
+            validation_callback_func,
+        ) in INCREMENTAL_TEST_CASES.items()
+    ],
+    ids=[test_name for test_name in INCREMENTAL_TEST_CASES],
+)
+def test_compact_partition_incremental(
+    s3_resource,
+    ds_mock_kwargs,
+    compaction_artifacts_s3_bucket,
+    test_name,
+    primary_keys,
+    in_sort_keys,
+    in_partition_keys,
+    column_names: List[str],
+    arrow_arrays: List[pa.Array],
+    rebase_source_partition_locator,
+    partition_values,
+    expected_result,
+    validation_callback_func,  # use and implement if you want to run additional validations apart from the ones in the test
+):
+
+    """
+    TODO Test Cases:
+    1. incremental w/wout round completion file
+    2. Backfill w/wout round completion
+    3. Rebase w/wout round completion file
+    4. Rebase then incremental (use same round completion file)
+    """
+    import deltacat.tests.local_deltacat_storage as ds
+    from deltacat.types.media import ContentType
+    from deltacat.storage import Partition, Stream
+    from deltacat.compute.compactor.compaction_session import compact_partition
+    from deltacat.storage.model.sort_key import SortKey
+    from deltacat.storage import (
+        PartitionLocator,
+    )
+    from deltacat.utils.placement import (
+        PlacementGroupManager,
+    )
+    from deltacat.compute.compactor import (
+        RoundCompletionInfo,
+    )
+
+    source_namespace = TEST_SOURCE_NAMESPACE
+    source_table_name = TEST_SOURCE_TABLE_NAME
+    source_table_version = TEST_SOURCE_TABLE_VERSION
+    destination_namespace = TEST_DESTINATION_NAMESPACE
+    destination_table_name = TEST_DESTINATION_TABLE_NAME
+    destination_table_version = TEST_DESTINATION_TABLE_VERSION
+    # setup
+    sort_keys = None
+    if in_sort_keys is not None:
+        sort_keys = [SortKey.of(sort_key["key_name"]) for sort_key in in_sort_keys]
+    partition_keys = None
+    if in_partition_keys is not None:
+        partition_keys = [
+            PartitionKey.of(
+                partition_key["key_name"], PartitionKeyType(partition_key["key_type"])
+            )
+            for partition_key in in_partition_keys
+        ]
+    # create the source table
+    ds.create_namespace(source_namespace, {}, **ds_mock_kwargs)
+    ds.create_table_version(
+        source_namespace,
+        source_table_name,
```

Comment:
This can be abstracted to a separated method named `setup_tables`.

Reply:
This get's kind of tricky with parametrize as we are using some of the parameters to construct the tables. Will need to see how best to accomplish this - using `indirect` may work - https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments

Reply:
Done @raghumdani